DataAccess/Repository/Interfaces/IRepository.cs

```diff
@@ -0,0 +1,19 @@
+﻿using DataAccess.Infrastructure;
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DataAccess.Repository.Interfaces
+{
+    public interface IRepository<TEntity>
+    {
+        Task<IReadOnlyCollection<TEntity>> GetAllAsync();
+        Task<IReadOnlyCollection<TEntity>> FilterAsync(Expression<Func<TEntity, bool>> predicate);
+        Task<TEntity> FindAsync(Expression<Func<TEntity, bool>> predicate);
+        Task CreateAsync(TEntity entity);
+        Task<TEntity> FindAsync(Guid id);
```

Comment:
not every entity would have guid as a key. Add a TKey param to interface as I mentioned before